docs/tooling-consolidation.md

```diff
@@ -0,0 +1,83 @@
+
+To achieve this we will be migrating the following (and associated tests) from aspnetcore-tooling to aspnetcore:
+```
+Microsoft.AspNetCore.Mvc.Razor.Extensions.Version1_X
+Microsoft.AspNetCore.Mvc.Razor.Extensions.Version2_X
+Microsoft.AspNetCore.Mvc.Razor.Extensions
+Microsoft.AspNetCore.Razor.Language
+Microsoft.AspNetCore.Razor.Tools
+Microsoft.CodeAnalysis.Razor
+Microsoft.NET.Sdk.Razor
+RazorPageGenerator
+````
+The following (and associated tests) will remain in aspnetcore-tooling:
+```
+Microsoft.AspNetCore.Razor.LanguageServer.Common
+Microsoft.AspNetCore.Razor.LanguageServer
+Microsoft.AspNetCore.Razor.OmniSharpPlugin.StrongNamed
+Microsoft.AspNetCore.Razor.OmniSharpPlugin
+Microsoft.AspNetCore.Razor.VSCode.Extension
+Microsoft.AspNetCore.Razor.VSCode
+Microsoft.CodeAnalysis.Razor.Workspaces
+Microsoft.CodeAnalysis.Remote.Razor
+Microsoft.VisualStudio.Editor.Razor
+Microsoft.VisualStudio.LanguageServerClient.Razor
+Microsoft.VisualStudio.LanguageServices.Razor
+Microsoft.VisualStudio.LiveShare.Razor
+Microsoft.VisualStudio.Mac.LanguageServices.Razor
+Microsoft.VisualStudio.Mac.RazorAddin
+Microsoft.VisualStudio.RazorExtension
+RazorDeveloperTools
+rzls
+````
+
+Due to the separation of tooling projects, the new dependency flow will be:
+
+```
+runtime +--> aspnetcore +--> SDK
+        \               \
+         +-> extensions -+-> aspnetcore-tooling
+```
+## Known action items
+
+1. Migrate source with `git filter-branch`, preserving commit history
+2. Migration of pending PRs to aspnetcore after source migration.
+   1. Alternatively, we can freeze checkins during source migration especially since there isn't much source code that requires migration
+3. Build razor SDK in pre-build step along with RepoTools and install it locally before the main build in aspnetcore
+   1. Can this step be skipped if for devs who don't work in this area?
+4. Invert darc subscriptions so aspnetcore flows into aspnetcore-tooling
+
+## Considerations
+
+### Pinning Roslyn dependencies
+
+Due to runtime and tooling divergence of Roslyn packages, we may need to pin multiple versions of Roslyn dependencies for the migrated packages.
+
+### Assets
+
+The migrated projects are C# only so we will use existing infrastructure in aspnetcore to handle signing and publishing of these packages.
+
+### Testing
+
+The migrated projects are C# only so there are no test infrasture changes needed in aspnetcore. The existing vscode jest, and node tests will remain in aspnetcore-tooling
+
+### Release cycles
+
+Tooling release cycles is a cross product of .NET Core (2.1, 3.1, 5.0) and VS (vs-mac, vscode, vs) release cycles. However, the packages we are migrating are only released on .NET Core cycles so it should not add any complexity to servicing in aspnetcore. We will follow the existing branching and servicing strategies.
```

Comment:
Does this mean we are migrating all of the packages that ship with .net core, and leaving all the of the packages that ship with VS?

Reply:
Yea that's the plan. That's how we came up with the list below.